### DISCUSSION TITLE
Workflow 3.0 Create Definition Giving Error

### DISCUSSION THREAD

**Discussion by vsanghavi-conga** [elsa-workflows/elsa-core#3722] [Q&A]
Hi,

I am trying to setup workflow server v3 by following this Guide - https://v3.elsaworkflows.io/docs/installation/aspnet-apps-workflow-server. While I was able to setup server, but then I received below error when tried to Create a workflow. It's the same payload as mentioned in Guide link above.

![image](https://user-images.githubusercontent.com/122509094/220112322-85b1eaf9-49e9-45b4-be8c-64b2be1b900d.png)

1 Can someone help what I am missing here ?
2 To follow Guides (Server, Designer, Server + Designer) What is the ideal preview version ?

Regards,
Viraj

**Comment by sfmskywalker** [comment]
Hi Viraj,

I just gave it a shot and I can confirm the issue and will push a fix soon.

**Reply by sfmskywalker** [reply]
This has been fixed, but due to a breaking change in GitHub's update to use the latest SDK, the build pipeline isn't working at the moment. We're fixing it as we speak.

**Comment by vsanghavi-conga** [comment]
Hi @sfmskywalker , Today I tried with **3.0.0-rc1** packages and it resulted in a different error. I followed the instructions on this page to create a workflow - https://v3.elsaworkflows.io/docs/installation/aspnet-apps-workflow-server.

![image](https://github.com/elsa-workflows/elsa-core/assets/122509094/6643e32e-c82e-4418-b164-f8573345a529)

**Comment by vsanghavi-conga** [comment]
https://github.com/elsa-workflows/elsa-core/tree/v3-rc1/src/samples/aspnet/Elsa.Samples.Webhooks.WorkflowServer

I tried running above sample app to debug and it seems it's failing to deserialize the Http request body.

![ElsaV3_DeserializationError](https://github.com/elsa-workflows/elsa-core/assets/122509094/463b2c6b-02bd-4bf7-97e1-28d913f45b47)

**Reply by sfmskywalker** [reply]
You're absolutely right, the model has changed since I wrote that documentation - I'll get it updated. Thanks for letting me know about it! 🙏🏻

**Reply by sfmskywalker** [reply]
The correct JSON payload is this:

```json
{
    "model": {
        "name": "Hello World",
        "root": {
            "type": "Elsa.WriteLine",
            "text": "Hello World!"
        }
    },
    "publish": true
}
```

**Reply by sfmskywalker** [reply]
Docs updated.

**Reply by vsanghavi-conga** [reply]
Thanks a lot for the quick reply.
